src/main/java/net/dries007/tfc/compat/jei/TFCJEIPlugin.java

```diff
@@ -293,5 +335,46 @@ public void register(IModRegistry registry)
         registry.addRecipeClickArea(GuiCrucible.class, 139, 100, 10, 15, ALLOY_UID);
         registry.addRecipeClickArea(GuiCrucible.class, 82, 100, 10, 15, METAL_HEAT_UID);
         registry.addRecipeClickArea(GuiFirePit.class, 79, 37, 18, 10, HEAT_UID);
+
+        // Fix inventory tab overlap see https://github.com/TerraFirmaCraft/TerraFirmaCraft/issues/646
+        registry.addAdvancedGuiHandlers(new TFCGuiAreas<>(GuiInventory.class));
+        registry.addAdvancedGuiHandlers(new TFCGuiAreas<>(GuiCalendar.class));
+        registry.addAdvancedGuiHandlers(new TFCGuiAreas<>(GuiNutrition.class));
+        registry.addAdvancedGuiHandlers(new TFCGuiAreas<>(GuiSkills.class));
+    }
+
+    private static class TFCGuiAreas<T extends GuiContainer> implements IAdvancedGuiHandler<T>
```

Comment:
This doesn't make any sense to be an inner class, nor does the name imo. Suggested: `TFCInventoryGuiHandler` (matches the name of the container)